src/Agent.Worker/TaskCommandExtension.cs

```diff
@@ -573,6 +577,7 @@ internal static class TaskSetVariableEventProperties
     internal static class TaskCompleteEventProperties
     {
         public static readonly String Result = "result";
+        public static readonly String Code = "code";
```

Comment:
As we discussed, for setting the task failure specific codes for different use cases customer issues/infra issues etc.